unit_tests/sources/declarative/retrievers/test_simple_retriever.py

```diff
@@ -1644,3 +1645,162 @@ def _mock_paginator():
     paginator.get_request_body_data.__name__ = "get_request_body_data"
     paginator.get_request_body_json.__name__ = "get_request_body_json"
     return paginator
+
+
+def test_fetch_one_simple_pk():
+    """Test fetch_one with a simple string primary key."""
+    requester = MagicMock()
+    requester.get_path.return_value = "posts"
+
+    response = requests.Response()
+    response.status_code = 200
+    response._content = json.dumps({"id": "123", "title": "Test Post"}).encode("utf-8")
+
+    requester.send_request.return_value = response
+
+    record_selector = MagicMock()
+    record_selector.select_records.return_value = [
+        Record(data={"id": "123", "title": "Test Post"}, stream_name="posts", associated_slice=None)
+    ]
```

Comment:
The way this test is written, we aren't actually testing the selector/extractor logic. Most extractor/selector are defined to extract a list of objects (Stripe almost certainly does it this way)

But here we just mock the single record response, and we mock calling `select_records()` so in a real world use case, this would fail since we'll attempt to extract `response.data` which doesn't exist. But all this is skipped over due to all the mocking